File: go/172/main.go
package main

import (
	"fmt"
)

func countEvenOdd(nums []int) (int, int) {
    even, odd := 0, 0
    for _, num := range nums {
        if num%2 == 0 {
            even++
        } else {
            odd++
        }
    }
    return even, odd
}

func main() {
    e, o := countEvenOdd([]int{1, 2, 3, 4, 5})
    fmt.Printf("偶数: %d, 奇数: %d\n", e, o) // 偶数: 2, 奇数: 3

    e, o = countEvenOdd([]int{0, 2, 4, 6})
    fmt.Printf("偶数: %d, 奇数: %d\n", e, o) // 偶数: 4, 奇数: 0

    e, o = countEvenOdd([]int{1, 3, 5, 7})
    fmt.Printf("偶数: %d, 奇数: %d\n", e, o) // 偶数: 0, 奇数: 4
}

File: python/172/main.py
def two_sum_indices(nums, target):
    res = []
    n = len(nums)
    for i in range(n):
        for j in range(i + 1, n):
            if nums[i] + nums[j] == target:
                res.append([i, j])
    return res


# テスト例
print(two_sum_indices([2, 7, 11, 15], 9))  # [[0,1]]
print(two_sum_indices([1, 2, 3, 4, 3], 6))  # [[1,3], [2,4]]
print(two_sum_indices([3, 3, 3], 6))  # [[0,1], [0,2], [1,2]]
